### PR TITLE
Improved tests

### PR DESCRIPTION
**Pull request by twerkmeister**
can be merged after clarify command is fixed in rasa
* fixed ordering in disambiguation/negation tests
* reinstated add_card flow that was missing for disambiguation / negation tests
* made some tests less strict when it comes to slots, removing checks for exact value sets when a range of similar values works
* quoted yes and no messages in yaml as they might be translated to True/False if one is not careful with the parser

### CODE REVIEW

File: e2e_tests/passing/happy_path/user_books_a_restaurant.yml [hunk 1]
```diff
@@ -5,13 +5,11 @@ test_cases:
       - slot_was_set:
           - book_restaurant_name_of_restaurant: Xaigon
           - book_restaurant_number_of_people: "4"
-          - book_restaurant_date: tonight
```
Comment (tabergma) [inline e2e_tests/passing/happy_path/user_books_a_restaurant.yml:8]: You could just leave those slots empty. E.g.
```
- book_restaurant_date
- book_restaurant_time
```
That should also work.

File: e2e_tests/passing/happy_path/user_books_a_restaurant_with_an_alternative_time.yml [hunk 2]
```diff
@@ -5,21 +5,17 @@ test_cases:
       - slot_was_set:
           - book_restaurant_name_of_restaurant: Xaigon
           - book_restaurant_number_of_people: "4"
-          - book_restaurant_date: tonight
```
Comment (tabergma) [inline e2e_tests/passing/happy_path/user_books_a_restaurant_with_an_alternative_time.yml:8]: Same as above, remove the value from the slot.

File: e2e_tests/passing/happy_path/user_books_a_restaurant_with_their_own_alternative.yml [hunk 3]
```diff
@@ -11,15 +11,13 @@ test_cases:
       - utter: utter_ask_book_restaurant_alternative_dummy
       - user: 8pm is not good, how about a bit later then, like 9pm?
       - utter: utter_ask_confirm_slot_correction
-      - user: yes
-      - slot_was_set:
-          - book_restaurant_time: 9pm
```
Comment (tabergma) [inline e2e_tests/passing/happy_path/user_books_a_restaurant_with_their_own_alternative.yml:16]: Same here, leave value empty.

File: e2e_tests/passing/happy_path/user_fills_resetting_slot.yml [hunk 4]
```diff
@@ -20,11 +20,3 @@ test_cases:
       - utter: utter_can_do_something_else
       - user: Bye then
       - utter: utter_goodbye
-      - user: I would like to confirm my account
```
Comment (tabergma) [inline e2e_tests/passing/happy_path/user_fills_resetting_slot.yml:23]: Isn't this part here for checking that the slot values are reset and that they need to be filled again?

File: data/flows/patterns.yml [hunk 5]
```diff
@@ -25,11 +25,13 @@ flows:
     steps:
       - action: action_trigger_chitchat
 
+  # using chitchat here so that intentless is used for better testability
```
Comment (tabergma) [inline data/flows/patterns.yml:28]: does this mean that some of the failing tests are now passing? if yes, should we move them?

**Review by tabergma — CHANGES_REQUESTED** [review]

**Comment by tabergma** [comment]
Is your branch up to date? Seems like the actions were not executed successfully. There are some recent changes for that on main.

File: e2e_tests/passing/happy_path/user_books_a_restaurant.yml (hunk 1 above)
Comment (twerkmeister) [reply e2e_tests/passing/happy_path/user_books_a_restaurant.yml:8]: Ah good, didn't know it worked to have both dicts and strings in a yaml list 👍

File: e2e_tests/passing/happy_path/user_fills_resetting_slot.yml (hunk 4 above)
Comment (twerkmeister) [reply e2e_tests/passing/happy_path/user_fills_resetting_slot.yml:23]: can we save the steps here and just whether the slots in question were set to None?

**Comment by tabergma** [comment]
should we wait for the next rasa release and then update the rasa version? and should we move the flaky test maybe to a folder could "flaky"?

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+## Description
+
+## TODOs
+[ ] checked compared flaky tests with the [known list of flaky tests steps](https://www.notion.so/rasa/Flaky-E2E-Test-Steps-63864d3d8c7b4427a0f3df8052e39f21)
```
Comment (tabergma) [inline .github/pull_request_template.md:4]:
```suggestion
[ ] compared flaky tests with the [known list of flaky tests steps](https://www.notion.so/rasa/Flaky-E2E-Test-Steps-63864d3d8c7b4427a0f3df8052e39f21)
```

**Review by tabergma — APPROVED** [review]
Looks good! Just one minor suggestion 💯

**Pull request merged**